tasks/cycles/cycles.js

```diff
@@ -22,14 +32,51 @@ function calculateDiscount(redemption) {
  * implement factorial algorithm using for, while, do..while operators
  */
 {
-  const i = 10; //10! = 3628800
+ const i =10; //10! = 3628800
+ var result = 1;
+  
+  for(let a = 1; a <= 10; a ++){
+    result = result * a;
+    console.log(result); 
+  }
+  
+}
+
+{
+var result = 1;
+let i = 0;
+
+  while(i < 10){
+    i = i + 1;
+    result = result * i;
+    console.log(result);
+  }
+}
+
+{
+  var result = 1;
+  let i = 0;
+
+  do {
+    i = i + 1;
+    result = result * i;
+    console.log(result)
+  }
+  while(i < 10);
+
 }
 
 /**
  * return concatenated string from an array of substring
  */
 {
   const substr = ["I", " love", " JS"];
+
+  for (let x in substr){
+  }
```

Comment:
you do not need console logs below since it should be done in the scope of the cycle:
let sum =''
for (let x of substr) {
sum = sum + x; OR sum +=x;
}
console.log(sum);